src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -290,11 +295,15 @@ RadixContainer<T> materialize_input(const std::shared_ptr<const Table>& in_table
   // Create histograms per chunk
   histograms.resize(chunk_count);
 
+  auto radix_container_node_positions = std::vector<std::optional<NodeID>>(chunk_count, std::nullopt);
```

Comment:
It seems that you set this to `UNKNOWN_NODE_ID` quite often (always, except for lines 440-441), could it be a better default value than `std::nullopt`? Do you even need optionals here, or could `merge_node_placements` be changed to not take `std::optional` either, but check for `UNKNOWN_NODE_ID` instead?